aten/src/THC/THCTensorMathReduce.cuh

```diff
@@ -625,7 +627,7 @@ THC_transformReduceInnermostDimIndex(THCState *state,
   }
   unsigned row_size = THCTensor_size(state, src, ndim - 1);
 
-  dim3 threads(16, 32);
+  dim3 threads(512);
```

Comment:
Hardcoding this to 512 might not be a good idea. What if the number of inputs you're reducing for every output is 32? You'll have 93% of your threads doing nothing.